### DISCUSSION TITLE
nginx config for nuxt on centos7

### DISCUSSION THREAD

**Discussion by ashkannazari248** [nuxt/nuxt#7741] [General]
hey
I have vps with centos7 operating system. I want to set up a site with nuxt on the server. I'm having trouble configuring nginx. I used the following code in the etc / nginx / nginx.conf file to configure nginx, but I get a 502 error

nginx config :

```nginx
map $sent_http_content_type $expires {
    "text/html"                 epoch;
    "text/html; charset=utf-8"  epoch;
    default                     off;
}

server {
    listen          80;            
    server_name     your-domain;    
    gzip            on;
    gzip_types      text/plain application/xml text/css application/javascript;
    gzip_min_length 1000;
    location / {
        expires $expires;

        proxy_redirect                      off;
        proxy_set_header Host               $host;
        proxy_set_header X-Real-IP          $remote_addr;
        proxy_set_header X-Forwarded-For    $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto  $scheme;
        proxy_read_timeout          1m;
        proxy_connect_timeout       1m;
        proxy_pass                          http://127.0.0.1:5000; 
    }
}
```

Also I can not access nuxt with my_ip: 5000.

I changed the port from 3000 to 5000.

My site is universal

I run the program with the following code:

"pm2 start npm -- start"

project folder : '/usr/share/nginx/html/myProjectFolder'

**Comment by pi0** [comment]
Hi @ashkannazari248. First congratulations that you were the first one tried `discussions` on nuxt :+1: :smile: 

I guess your nginx deployment should be almost fine but the issue is probably related to pm2 setup. Can you try `curl -v http://localhost:5000` on server? It probably will show more details on issue.


> server_name     your-domain;  

this should be either removed, set to `default` or your website domain name. Otherwise other server rules may wrongly match.

**Reply by ashkannazari248** [reply]
Hello . Yes. With the command you said, I will receive the information of the index page. Apparently the project is running well, but I see only a 502 error when I open the site!
When I run the project with the ```npm run dev``` command, I can not see anything with ```myDomain: 5000``` except error 502!

**Reply by ashkannazari248** [reply]
it is true ?
```
map $sent_http_content_type $expires {
    "text/html"                 epoch;
    "text/html; charset=utf-8"  epoch;
    default                     off;
}

server {
    listen          80;             
    gzip            on;
    gzip_types      text/plain application/xml text/css application/javascript;
    gzip_min_length 1000;

    location / {
        expires $expires;

        proxy_redirect                      off;
        proxy_set_header Host               $host;
        proxy_set_header X-Real-IP          $remote_addr;
        proxy_set_header X-Forwarded-For    $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto  $scheme;
        proxy_read_timeout          1m;
        proxy_connect_timeout       1m;
        proxy_pass                          http://127.0.0.1:5000;
    }
}
```
